tests/components/oncue/test_config_flow.py

```diff
@@ -138,3 +139,53 @@ async def test_already_configured(hass: HomeAssistant) -> None:
 
     assert result2["type"] is FlowResultType.ABORT
     assert result2["reason"] == "already_configured"
+
+
+async def test_reauth(hass: HomeAssistant) -> None:
+    """Test reauth flow."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_USERNAME: "any",
+            CONF_PASSWORD: "old",
+        },
+    )
+    config_entry.add_to_hass(hass)
+    config_entry.async_start_reauth(hass)
+    await hass.async_block_till_done()
+    flows = hass.config_entries.flow.async_progress_by_handler(DOMAIN)
+    assert len(flows) == 1
+    flow = flows[0]
+
+    with patch(
+        "homeassistant.components.oncue.config_flow.Oncue.async_login",
+        side_effect=LoginFailedException,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            flow["flow_id"],
+            {
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["errors"] == {"password": "invalid_auth"}
+
+    with (
+        patch("homeassistant.components.oncue.config_flow.Oncue.async_login"),
+        patch(
+            "homeassistant.components.oncue.async_setup_entry",
+            return_value=True,
+        ) as mock_setup_entry,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            flow["flow_id"],
+            {
+                CONF_PASSWORD: "test-password",
```

Comment:
Do we need to verify this new password percolates into the config_entry?

Reply:
b9c48d92998ee7b80696b9499286636fa8331787